packages/midi/midi.mjs

```diff
@@ -66,8 +67,57 @@ function getDevice(output, outputs) {
 // Pattern.prototype.midi = function (output: string | number, channel = 1) {
 Pattern.prototype.midi = function (output) {
   if (!supportsMidi()) {
-    throw new Error(`🎹 WebMidi is not enabled. Supported Browsers: https://caniuse.com/?search=webmidi`);
+    console.log(`🎹 WebMidi is not enabled. Supported Browsers: https://caniuse.com/?search=webmidi`);
+    return this.onTrigger((time, hap) => {
+      hap.ensureObjectValue();
+
+      const timingOffset = WebMidi.time - getAudioContext().currentTime * 1000;
+      time = time * 1000 + timingOffset;
+      // destructure value
+      const { note, nrpnn, nrpv, ccn, ccv, midichan = 1 } = hap.value;
+      const velocity = hap.context?.velocity ?? 0.9; // TODO: refactor velocity
+      const duration = hap.duration.valueOf() * 1000 - 100;
+
+      if (note) {
+        const midiNumber = noteToMidi(note);
+        setTimeout(() => {
```

Comment:
is there a way to avoid setTimeout? the timing of setTimeout is not to be trusted

Reply:
I'll see what I can do! it wast the first thing that worked for keeping notes more of less on the grid.